corelib-search/src/main/java/eu/europeana/corelib/search/impl/WebMetaInfo.java

```diff
@@ -192,14 +190,19 @@ private static String generateHashCode(String wrId, String recordId) {
      * a reference to IIIF (see ticket EA-992)
      * @param bean
      */
-    private static void addReferencedByIIIF(FullBean bean) {
+    private static void addReferencedByIIIF(FullBean bean, boolean manifestAddUrl, String api2BaseUrl) {
         // tmp add timing information to see impact
         long start = System.nanoTime();
         if (isNewsPaperRecord(bean) && !hasReferencedBy(bean) && bean.getAggregations() != null) {
             // add to all webresources in all aggregations
             for (Aggregation a : bean.getAggregations()) {
                 for (WebResource wr : a.getWebResources()) {
-                    String iiifId = "https://iiif.europeana.eu/presentation" + bean.getAbout() + "/manifest";
+                    String iiifId;
+                    if(manifestAddUrl){
+                        iiifId= "https://iiif.europeana.eu/presentation" + bean.getAbout() + "/manifest?recordApi="+api2BaseUrl;
+                    }else {
+                        iiifId = "https://iiif.europeana.eu/presentation" + bean.getAbout() + "/manifest";
+                    }
```

Comment:
This will work fine, but slightly more elegant would be to use 
`                    String iiifId = "https://iiif.europeana.eu/presentation" + bean.getAbout() + "/manifest";
                    if (manifestAddUrl) {
                        iiifId = iiifId + "/manifest?recordApi=" + api2BaseUrl;
                    }`
because then you don't have to duplicate the "https://iiif.europeana.eu/presentation" string.